### PR TITLE
new gcp jit command

### PR DESCRIPTION
**Pull request by kferrone**
## Describe Changes

New gcp jit command that retreives a gcp token from the duplo api. 

## Link to Issues  

https://app.clickup.com/t/8655600/DUPLO-15865

## PR Review Checklist  

- [x] Thoroughly reviewed on local machine.
- [ ] Have you added any tests
- [x] Make sure to note changes in Changelog

___

### **PR Type**
enhancement, documentation

___

### **Description**
- Introduced a new `gcp` command in `jit.py` to retrieve GCP JWT tokens for authentication with the GCP API.
- Implemented caching mechanism to store and retrieve GCP credentials efficiently.
- Added usage examples for setting environment variables for the `gcloud` CLI.
- Updated `CHANGELOG.md` to document the new GCP JIT command and its usage.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>jit.py</strong><dd><code>Add GCP JWT token retrieval command with caching</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/duplo_resource/jit.py

<li>Added a new <code>gcp</code> command to retrieve GCP JWT tokens.<br> <li> Implemented caching logic for GCP credentials.<br> <li> Provided usage examples for setting environment variables.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/duploctl/pull/116/files#diff-7e55cc23580ca0ed29a44b891d834a6cf3251a2a5a785f6291a4efe95b54cdf6">+41/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update changelog with GCP JIT command details</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

CHANGELOG.md

<li>Documented the addition of the GCP JIT command.<br> <li> Updated the changelog with example CLI usage.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/duploctl/pull/116/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-15865 GCP JIT](https://app.clickup.com/t/8655600/DUPLO-15865)

**Review by duplocloud-matt — APPROVED** [review]
lgtm!

File: src/duplo_resource/jit.py [hunk 1]
```diff
@@ -47,6 +47,47 @@ def token(self) -> dict:
...
     return {"token": self.duplo.token}
+  
+  @Command()
+  def gcp(self, nocache: bool = None) -> dict:
+    """GCP Access Token
+    
+    Get the GCP JWT token for the current user. This is the token that is used to authenticate with the GCP API. 
+
+    Example: Using for gcloud cli access
+      Here is how to set the needed environment variables for the gcloud cli.
```
Comment (gk-duplo) [inline src/duplo_resource/jit.py:58]: This works only for admin user. Should we mention it?

**Review by gk-duplo — APPROVED** [review]
Approved with small comment

File: src/duplo_resource/jit.py (hunk 1 above)
Comment (kferrone) [reply src/duplo_resource/jit.py:58]: Yeah. I also wonder why even have the tenant id in the url?

**Pull request merged**